### PR TITLE
[ui, deployments] Mirage Actively Deploying Job and Deployment Integration Tests

### PR DESCRIPTION
**Pull request by philrenaud**
- Adds a Mirage job with a fairly complex active deployment: 
image

- Adds integration tests for the job status panel with an active deployment

### CODE REVIEW

File: ui/app/components/job-status/allocation-status-block.hbs [hunk 1]
```diff
@@ -13,6 +13,9 @@
           @label="View allocation"
         >
           {{#if (and (eq @status "running") (not @steady))}}
+            {{#if (eq @canary "canary")}}
+              <span class="alloc-canary-indicator" />
+            {{/if}}
```
Comment (philrenaud) [inline ui/app/components/job-status/allocation-status-block.hbs:18]: Side-effect: we weren't showing the canary indicator for running, non-steady, ungrouped allocs. Now we are.

File: ui/mirage/scenarios/default.js [hunk 2]
```diff
@@ -86,6 +86,76 @@ function smallCluster(server) {
     type: 'service',
     activeDeployment: true,
   });
+
+  //#region Active Deployment
```
Comment (philrenaud) [inline ui/mirage/scenarios/default.js:90]: New default (small cluster) job to show off the deployment panel

File: ui/app/components/job-status/allocation-status-block.hbs [hunk 3]
```diff
@@ -13,6 +13,9 @@
           @label="View allocation"
         >
           {{#if (and (eq @status "running") (not @steady))}}
+            {{#if (eq @canary "canary")}}
```
Comment (ChaiWithJai) [inline ui/app/components/job-status/allocation-status-block.hbs:16]: `question`: Where are you injecting this argument?

**Review by ChaiWithJai — APPROVED** [review]
LGTM

File: ui/app/components/job-status/allocation-status-block.hbs (hunk 3 above)
Comment (philrenaud) [reply ui/app/components/job-status/allocation-status-block.hbs:16]: This is injected in the nested `each-in` blocks of allocation-status-row.hbs: https://github.com/hashicorp/nomad/blob/f64a0b51272e9a6bce9cff09bd2596a673a7cc0e/ui/app/components/job-status/allocation-status-row.hbs#L11-L18

**Pull request merged**